src/common-libs/auth/auth-checker.ts

```diff
@@ -0,0 +1,43 @@
+import { AuthChecker } from 'type-graphql';
+import { getLoggedInParticipant, getRoleGrantsForUser } from '.';
+import { createRootContextFromRequest } from '../../common-utils/logging';
+import Context from '../../domain-services/Context';
+
+export const authChecker: AuthChecker<Context> = async (
```

Comment:
the majority of use cases will need to check more than just whether a user has a global role... in which case supplying role names is insufficient, as you also need additional context (such as the ID of object in question).

Beyond this, authorization checking should always be done with respect to permissions, not roles. Otherwise changing the permitted actions of any particular role means updating every endpoint rather than simply removing / adding a permission.

Given that you've mostly ported `calculatePermissionsFromRolesGrant`, is there any particular reason why you're not just using that and `hasRequiredPermissions`?

Reply:
This decorator is now removed and only `@Permission` decorator is left.